velox/exec/tests/QueryAssertionsTest.cpp

```diff
@@ -343,4 +343,18 @@ TEST_F(QueryAssertionsTest, multiFloatColumnWithNonUniqueKeys) {
       "971 extra rows, 971 missing rows");
 }
 
+TEST_F(QueryAssertionsTest, nullDecimalValue) {
+  auto shortDecimal = makeRowVector({
+      makeNullableShortDecimalFlatVector(
+          {std::nullopt},
+          SHORT_DECIMAL(5, 2))});
+  EXPECT_TRUE(assertEqualResults({shortDecimal}, {shortDecimal}));
+
+  auto longDecimal = makeRowVector({
+      makeNullableLongDecimalFlatVector(
+          {std::nullopt},
+          LONG_DECIMAL(20, 2))});
+  EXPECT_TRUE(assertEqualResults({longDecimal}, {longDecimal}));
+}
```

Comment:
For completeness, could you also add a test case where assertEqualResults fails with two decimal null vector of different precisions and scales?

Reply:
Done.